src/services/userServices.js

```diff
@@ -14,19 +15,37 @@ const authorizeUser = () => {
   return returnMethod;
 };
 
+const getLocaleFromCookie = () => {
+  try {
+    const supportedLocales = process.env.REACT_APP_CONFIG_SERVICE_SUPPORTED_LOCALES.split(',');
+    const cookieValue = Cookie.get(process.env.REACT_APP_CONFIG_SERVICE_LOCALES_COOKIE);
+    if (cookieValue && supportedLocales.indexOf(cookieValue) > -1) {
+      const majorLocaleValue = cookieValue.split('_')[0];
+      return { value: majorLocaleValue, key: cookieValue };
+    }
+    return null;
+  } catch (e) {
+    if (!helpers.TEST_MODE) {
+      console.warn(`Unable to parse locale ${e.message}`);
+    }
+    return null;
+  }
+};
+
 const getLocale = () => {
-  const locale = {
+  const defaultLocale = {
     value: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_DEFAULT_LNG,
-    key: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_DEFAULT_LNG_DESC
+    key: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_DEFAULT_LNG_KEY
   };
+  const cookieLocale = getLocaleFromCookie();
+  const locale = cookieLocale || defaultLocale;
 
   return new Promise(resolve => {
     if (locale) {
       return resolve({
-        data: locale
+        data: getLocaleFromCookie() || defaultLocale
       });
     }
-
     return resolve({});
   });
 };
```

Comment:
so this gets weird... i've gone ahead and updated to use `en-US.json` b/c that's what we discussed, and I think it's the most readable...and will totally work....

BUTTTT... i18nnext will by default translate these four/five letter codes into "fallback" two letter codes... so it will request both `en.json` and `en-US.json` as the primary once it does not find `en.json`... saw that logic in the debugger here after testing this:
https://github.com/i18next/i18next/blob/270904f6369ee9bbda059c3186fcea7baf9eb15d/src/LanguageUtils.js#L33

All this being said... i still prefer to "key" on the four/five letter codes. Better specificity for future possibilities of supporting those minor locales. Also... having a "major" locale fallback file could also be a thing and just supported out of the box w/ this kind of configuration. Pretty cool it does this!

Reply:
Have 2 easy solutions 
Either:
1. place an empty `en.json` file with an empty JSON object in there... that way the fallback happens... or 
2. reverse it... so `en.json` is still populated and `en-US.json` is an empty JSON object... 

Either way works and gets us easily past the load... I'll add it into a review commit

Reply:
woooot